util/util.go

```diff
@@ -380,3 +380,13 @@ func HasOwner(refList []metav1.OwnerReference, apiVersion string, kinds []string
 
 	return false
 }
+
+// IsPaused returns true if the object has the `pause-reconciler` annotation.
+func IsPaused(v metav1.Object) bool {
```

Comment:
@vincepri what about changing this logic so an object is suspended if the reconciliation is paused for the object itself or for the cluster the objects belong to (pausing the cluster pauses all)?
This will greatly  simplify clusterctl move or other actions where you want to freeze a whole cluster

Reply:
If we want to support only having to set on a Cluster resource, then maybe we should consider a high level field on the Cluster rather than an annotation.

Reply:
Agreed, might as well be explicit and have a field

Reply:
The use case in the linked issue was also to allow for specific object to be stopped from reconciling